pgjdbc/src/main/java/org/postgresql/jdbc/PgPreparedStatement.java

```diff
@@ -626,9 +627,15 @@ public void setObject(int parameterIndex, Object in, int targetSqlType, int scal
         setString(parameterIndex, castToString(in), Oid.BPCHAR);
         break;
       case Types.VARCHAR:
-      case Types.LONGVARCHAR:
         setString(parameterIndex, castToString(in), getStringType());
         break;
+      case Types.LONGVARCHAR:
+        if (in instanceof InputStream) {
+          preparedParameters.setText(parameterIndex, (InputStream)in);
```

Comment:
Pretty sure we can guarantee that there will be a UTF server. But that can't be guaranteed in the wild. We just set the encoding of the connection. Is that not possible ?